src/table/table.scss

```diff
@@ -4,383 +4,373 @@
-    }
 
-    &.iui-tables-row-new {
-      font-weight: $iui-font-weight-semibold;
+      + .iui-selected {
+        border-bottom-color: transparent;
+        @include themed {
+          border-top-color: rgba(t(iui-color-foreground-primary-rgb), t(iui-opacity-4));
+        }
+      }
 
-      > .iui-tables-filename {
-        position: relative;
+      &:last-child {
+        @include themed {
+          border-bottom-color: t(iui-color-foreground-primary);
+        }
       }
 
-      > .iui-tables-filename::before {
+      + :not(.iui-selected),
+      + .iui-expanded-content + :not(.iui-selected) {
         @include themed {
-          color: t(iui-color-foreground-positive);
+          border-top-color: t(iui-color-foreground-primary);
         }
+      }
+
+      + .iui-expanded-content {
+        @include themed {
+          border-color: transparent t(iui-color-foreground-primary);
+        }
+      }
+    }
+    // #endregion Selection
+
+    // #region Statuses & states
+    &.iui-new {
+      font-weight: $iui-font-weight-semibold;
+      
+      > .iui-main-column::before {
         content: '•';
         position: absolute;
         left: 0;
         font-size: $iui-font-size-title;
+        @include themed {
+          color: t(iui-color-foreground-positive);
+        }
       }
     }
 
-    &.iui-tables-row-disabled {
+    &.iui-disabled,
+    &.iui-disabled + .iui-expanded-content {
+      font-style: italic;
+      cursor: not-allowed;
       @include themed {
         color: t(iui-text-color-muted);
       }
-      font-style: italic;
-      cursor: not-allowed;
 
-      &:hover {
-        @include themed {
-          background-color: t(iui-color-background-1);
-        }
-
-        > .iui-tables-slot > .iui-tables-menu {
-          visibility: hidden;
-        }
+      &:hover > .iui-slot > .iui-more-options {
+        visibility: hidden;
       }
 
-      > .iui-tables-cell {
-        > img,
-        > svg,
-        > .iui-user-icon,
-        > .iui-user-icon > img {
-          filter: $iui-icons-color-multicolor-disabled;
-        }
+      img,
+      svg:not(.iui-radial),
+      .iui-user-icon {
+        filter: $iui-icons-color-multicolor-disabled;
       }
     }
+
+    &.iui-positive {
+      @include iui-table-row-status($status:positive);
+    }
+    
+    &.iui-warning {
+      @include iui-table-row-status($status:warning);
+    }
+
+    &.iui-negative {
+      @include iui-table-row-status($status:negative);
+    }
+    // #endregion Statuses & states
   }
 
-  .iui-tables-message-container {
+  // Empty & loading states
+  > .iui-table-empty {
+    text-align: center;
+    padding: $iui-xl;
     @include themed {
       color: t(iui-text-color-muted);
       background-color: t(iui-color-background-2);
     }
+  }
+}
+
+@mixin iui-table-cell {
+  align-items: center;
+  display: flex;
+  flex-grow: 1;
+  min-width: $iui-xxl;
+  min-height: $iui-baseline * 5;
+  padding-left: $iui-m;
+  overflow: hidden;
```

Comment:
Doesn't really work:
![image](https://user-images.githubusercontent.com/36186912/121472249-9a580000-c9c9-11eb-960e-3205cb91a8c0.png)

Reply:
I think it will work if user provides a width

Reply:
I tried applying `width: 172px; text-overflow: ellipsis;` and am not getting truncation.
![Screen Shot 2021-06-10 at 8 46 32 AM](https://user-images.githubusercontent.com/849817/121527398-5ca6a780-c9c8-11eb-8260-ea63c15ed35f.png)

Reply:
Ah right, text-overflow also needs `white-space: nowrap`

Reply:
So `overflow: hidden` is not really doing anything since the height is not constrained (which is what `nowrap` does).
![image](https://user-images.githubusercontent.com/9084735/121527917-eeaeb000-c9c8-11eb-98a0-937bd6ef2805.png)

We should just remove it or constrain the height in some way

Reply:
Removed